### PR TITLE
fix(parser): fix annexB behaviour (webcompat) of catch param

### PR DESCRIPTION
**Pull request by 3cp**
ECMA B.3.4 syntax error if catch param occurs in var declared names of catch block, unless catch param is binding identifier

### CODE REVIEW

File: scripts/run-test262.ts [hunk 1]
```diff
@@ -14,24 +14,7 @@ function loadList(filename: string) {
 }
 
 const unsupportedFeatures = new Set(loadList('unsupported-features.txt'));
-const whitelist = [
-  ...loadList('whitelist.txt'),
-  // Can't construct these regexps on Node.js v16
-  ...(process.version.startsWith('v16.')
```
Comment (3cp) [inline scripts/run-test262.ts:20]: Nodejs v16 is removed from CI, and removed from package.json engine.

File: src/common.ts [hunk 2]
```diff
@@ -724,12 +724,8 @@ export function addVarName(
...
       }
     }
-    if (value & (BindingKind.CatchIdentifier | BindingKind.CatchPattern)) {
-      if (
-        (value & BindingKind.CatchIdentifier) === 0 ||
-        (context & Context.OptionsWebCompat) === 0 ||
-        context & Context.Strict
-      ) {
+    if (value & BindingKind.CatchIdentifierOrPattern) {
+      if ((context & Context.OptionsWebCompat) === 0 || (value & BindingKind.CatchIdentifier) === 0) {
```
Comment (3cp) [inline src/common.ts:728]: When in webcompat mode (annexB), only fail if it's CatchPattern `catch ({message}) { var message }`.

Comment (fisker) [inline src/common.ts:728]: [unicorn/no-lonely-if](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-lonely-if.md)

**Review by fisker — APPROVED** [review]

File: src/common.ts [hunk 3]
```diff
@@ -724,14 +724,11 @@ export function addVarName(
...
-        (value & BindingKind.CatchIdentifier) === 0 ||
-        (context & Context.OptionsWebCompat) === 0 ||
-        context & Context.Strict
-      ) {
-        report(parser, Errors.DuplicateBinding, name);
-      }
+    if (
+      value & BindingKind.CatchPattern ||
+      (value & BindingKind.CatchIdentifier && (context & Context.OptionsWebCompat) === 0)
+    ) {
```
Comment (3cp) [inline src/common.ts:730]: The cleaned up if-logic is more readable.

Comment (fisker) [reply src/common.ts:730]: Did we duplicate the `value` check?

Comment (fisker) [reply src/common.ts:730]: Sorry... it's `&`..

**Review by fisker — APPROVED** [review]

File: src/common.ts [hunk 4]
```diff
@@ -724,10 +724,11 @@ export function addVarName(
         currentScope.scopeError = recordScopeError(parser, Errors.DuplicateBinding, name);
       }
     }
-    if (value & BindingKind.CatchIdentifierOrPattern) {
-      if ((context & Context.OptionsWebCompat) === 0 || (value & BindingKind.CatchIdentifier) === 0) {
-        report(parser, Errors.DuplicateBinding, name);
-      }
+    if (
+      value & BindingKind.CatchPattern ||
+      (value & BindingKind.CatchIdentifier && (context & Context.OptionsWebCompat) === 0)
```
Comment (fisker) [inline src/common.ts:729]: @3cp I didn't use bitmask much, but why are we using it for options? Shouldn't we use boolean instead?

Comment (3cp) [reply src/common.ts:729]: Context uses bitmask for all options, as options can be applied together.

**Pull request merged**